### ISSUE TITLE
Slow To Load Messages at Launch

### ISSUE THREAD

**Issue by Rov3r** [FossifyOrg/Messages#234]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

Android 14

### Affected device model

Samsung Galaxy S23 Ultra

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Open the app
2. App launches but nothing displayed in Messages view

### Expected behavior

Should be near instantaneous to load messages

### Actual behavior

The app launches and won’t load apps for 5-8 seconds on average.

### Additional information

This is installed on someone else’s phone. They installed it today as their carrier’s texting app is going away, and I recommended they get a messaging app from F-Droid. Upon the first launch, all messages were loaded in—no problem. However they complained to me that it takes a long time to be able to interact with messages compared to the carrier’s messaging app. I checked and confirmed it does take much longer to launch the Fossify Messages.

**Comment by Aga-C** [comment]
Do you know how many conversations are on the phone?

**Comment by Rov3r** [comment]
1. The whole conversation list
2. 44,000+
3. N/A

**Comment by Rov3r** [comment]
Yeah unfortunately they’ve been complaining about it to me so much that I gave up and said just go back to using the Samsung app. I really hope this does get fixed in the future. When I migrate to Android it would be nice to know there’s FOSS for messaging.

**Comment by jayb-g** [comment]
> Overall it feels very sluggish to use as compared to other SMS apps.

And this slow-ness isn't limited to the fossify Messages app. It (seemingly)delays loading of new items in the fossify Gallery app too.

I'm not using any other fossify apps but seems like something common in both(and other fossify) apps is causing this sluggish behavior.

**Comment by naveensingh** [comment]
How many files do you have? You might want to subscribe to the following issues:
 - https://github.com/FossifyOrg/Gallery/issues/54
 - https://github.com/FossifyOrg/Gallery/issues/284

**Comment by joshhansen** [comment]
I have 44,851 messages in 1,045 conversations from about 10 years of messaging history. Many operations in Fossify are quite slow:
1. Initial loading of main conversation list
2. Initial loading of an individual conversation
3. Setting a conversation to "Unread"
4. Conversation being marked as read after viewing it then returning to the conversations list
5. Subsequent re-loading of a conversation that has a new message added to it

The slowness is intolerable, I will have to switch to another app unfortunately

App version 1.1.5, Android 14, Samsung Galaxy S22

The name of this issue is too narrow, should probably be changed to: UI is unacceptably slow with high message count

**Comment by naveensingh** [comment]
The main conversation list is still slow, but chat messages should load faster now in [v1.5.0](https://github.com/FossifyOrg/Messages/releases/tag/1.5.0).

**Comment by salfter** [comment]
> The main conversation list is still slow, but chat messages should load faster now in [v1.5.0](https://github.com/FossifyOrg/Messages/releases/tag/1.5.0).

I just checked after having to wait for some conversations to load up.  I'm on that version, and it's still pretty slow.  It wasn't like this until maybe the past week or so, which suggests a regression has been introduced.

**Comment by naveensingh** [comment]
> I just checked after having to wait for some conversations to load up.  I'm on that version, and it's still pretty slow.  It wasn't like this until maybe the past week or so, which suggests a regression has been introduced.

- How many messages are there in that conversation?
- How many total conversations do you have?

Please also share your device info. Related: https://github.com/FossifyOrg/Messages/issues/574

**Comment by particlesurfer** [comment]
Apologies for making Dupe ticket. I did  not come across this one. 
This is still an ongoing issue in Dec 2025
😱😭

**Comment by naveensingh** [comment]
> This is still an ongoing issue in Dec 2025 😱😭

Oh, don't cry. It'll be fixed.

**Comment by particlesurfer** [comment]
> > This is still an ongoing issue in Dec 2025 😱😭
> 
> Oh, don't cry. It'll be fixed.

🤙😅🙌🫶
I love this application so much and you guys for making it!! after going through many apps. this is def paramount! Congrats and thanks SO much for the support!! looking forward to donating once I get paid!

**Comment by Rov3r** [comment]
Thank you for your attention to this issue. I have confidence you will address the issue at some point. Open source is often a thankless pursuit but projects like yours are exactly what the world needs. Not every project has endless resources, but I do plan to contribute at some point.

**Comment by APileOfSticks** [comment]
I began using FOSSIFY MSGS 2 days ago.  I have same issue on same phone model s23 ultra as reported in @Rov3r October 15 2024 original post.  I open app, no messages.  Messages load after about 15 second count.  I love this app.  Hope issue is resolved soon.

**Comment by prawn778** [comment]
Is this issue still being worked on or should I look for an alternative app?
5+ sec delay on opening app to get the message list (v1.8.0, Android v12)

Thanks

**Comment by jayb-g** [comment]
> Is this issue still being worked on or should I look for an alternative app?


Its not practical to use this app if you have large db of SMS 

Better to switch to [Quik](https://github.com/octoshrimpy/quik) until this is fixed

**Comment by prawn778** [comment]
OK - thanks
